pep-0517.txt

```diff
@@ -166,48 +166,71 @@ The build backend object is expected to have attributes which provide
 some or all of the following hooks. The common ``config_settings``
 argument is described after the individual hooks::
 
-  def get_build_requires(config_settings):
+  def build_sdist(source_directory, sdist_directory, config_settings):
+      ...
+
+MUST build an unpacked sdist from the ``source_directory`` and place it into the
+specified ``sdist_directory``. If the ``source_directory`` is already an
+unpacked sdist (as opposed to a VCS source tree) then it SHOULD simply copy the
```

Comment:
> - Do we specify how to tell what case it's handling?
> - What about cases which are neither? E.g. if I download a tarball from Github and unpack it, it is neither a VCS checkout (no .git directory), nor an sdist. Is it OK to error out? Should frontend tools try something else if this fails?

I think that's the backend job to try to detect which case it's handling, and it's completely fair of the backend to throw up it's hands if it's in a situation it was not designed to handle (like, being neither in a VCS checkoukt nor in an sdist).

Backends could then be improved to handle different situations. I don't think we can future proof this part (e.g. an export from github migth look different from an export from sourceforge, due to the inclusion or not of a top-level directory with an arbitrary name that might or might not include the repository name).